src/global_logic.js

```diff
@@ -1975,7 +1970,7 @@ module.exports.treatSupportAbility = function (totals, chara) {
                     continue;
                 case "envoy_meditation":
                     var elements = module.exports.checkNumberOfElements(totals);
```

Comment:
```javascript
var elements = Math.min(4, module.exports.checkNumberOfElements(totals));
```

then code below does not need to check elements >= 4 every times.
```javascript

totals[key]["normalBuff"] += (elements * 0.15);
totals[key]["DABuff"] += (elements * 0.10);
totals[key]["TABuff"] += (elements * 0.03);
```
for more clean up (hand write code here in comment form, not tested)

Reply:
👍